test/integration/patients/sidebar/action-sidebar.js

```diff
@@ -1515,6 +1515,184 @@ context('action sidebar', function() {
+          message: 'Test comment to be deleted.',
+        },
+        relationships: {
+          clinician: getRelationship(getCurrentClinician()),
+        },
+      }),
+      getComment({
+        attributes: {
+          edited_at: null,
+          created_at: testTsSubtract(2),
+          message: 'Another test comment to be deleted.',
+        },
+        relationships: {
+          clinician: getRelationship(getCurrentClinician()),
+        },
+      }),
+    ];
+
+    const testAction = getAction({
+      relationships: {
+        patient: getRelationship(testPatient),
+        state: getRelationship(stateTodo),
+        comments: getRelationship(testComments),
+      },
+    });
+
+    cy
+      .routesForPatientAction()
+      .routePatient(fx => {
+        fx.data = testPatient;
+
+        return fx;
+      })
+      .routePatientActions(fx => {
+        fx.data = [testAction];
+
+        return fx;
+      })
+      .routePatientFlows(fx => {
+        fx.data = [];
+
+        return fx;
+      })
+      .routeAction(fx => {
+        fx.data = testAction;
+
+        return fx;
+      })
+      .routeActionFiles(fx => {
+        fx.data = [];
+
+        return fx;
+      })
+      .routeActionComments(fx => {
+        fx.data = testComments;
+
+        return fx;
+      })
+      .visit(`/patient/${ testPatient.id }/action/${ testAction.id }`)
+      .wait('@routePatientActions')
+      .wait('@routePatientFlows')
+      .wait('@routeAction')
+      .wait('@routeActionFiles');
+
+    cy
+      .intercept('DELETE', '/api/comments/*', {
+        statusCode: 204,
+        body: {},
+      })
+      .as('routeDeleteComment');
+
+    cy
+      .get('[data-activity-region]')
+      .find('.comment__item')
+      .first()
+      .find('.js-edit')
+      .click();
+
+    cy
+      .get('[data-activity-region]')
+      .find('.js-delete')
+      .click();
+
+    cy
+      .get('.modal--small')
+      .find('.js-submit')
+      .click()
+      .wait('@routeDeleteComment');
+
+    cy
+      .get('.patient__list')
+      .find('.table-list__item .fa-comment')
+      .should('exist')
+      .next()
+      .should('contain', '1');
+
+    cy
+      .get('[data-activity-region]')
+      .find('.comment__item')
+      .first()
+      .find('.js-edit')
+      .click();
+
+    cy
+      .get('[data-activity-region]')
+      .find('.js-delete')
+      .click();
+
+    cy
+      .get('.modal--small')
+      .find('.js-submit')
+      .click()
+      .wait('@routeDeleteComment');
+
+    cy
+      .get('.patient__list')
+      .find('.table-list__item .fa-comment')
+      .should('not.exist');
+
+    cy
+      .get('.patient__list')
+      .find('.table-list__item .fa-comment')
+      .should('not.exist');
+
+    cy
+      .get('.sidebar')
+      .find('[data-comment-region]')
+      .as('postCommentRegion')
+      .find('.js-input')
+      .type('Test comment');
+
+    cy
+      .intercept('POST', '/api/actions/*/relationships/comments', {
```

Comment:
It's possible I suppose that the 204 is preventing the id, but I don't think it does..  we could check that though.

Reply:
The `204` code was the problem. Switched them to `201` and the response body worked 👍 thanks!